dask_sql/physical/utils/sort.py

```diff
@@ -6,12 +6,34 @@
 from dask_sql.utils import make_pickable_without_dask_sql, new_temporary_column
 
 
+def multi_col_sort(
```

Comment:
Maybe we can call it `fast_multi_col_sort` (or alike), because the other function also does a multi column sort in some way?

Reply:
I was thinking something closer to the original function name that also references the fact that this relies on dask-cudf's codepath (such as `cudf_apply_sort`) - how does that sound?

Reply:
I think I'm somewhere in-between these two opinions :)

I'm hoping that `sort_values()` to work for both pandas and cudf-backed data in the future, so it would probably be better to avoid using `cudf` in the function name.  Also, the current multi-column sort for pandas-backed data produces correct ordering, but it does not produce "proper" partitioning.  My understanding is that the percentiles of the first column are used to define new divisions, which can lead to some ugly partitioning when the number of partitions is not small relative to the cardinality of the first column.

Therefore, I personally feel that it makes the most sense to start by trying to call the "proper" sort, and then fall back on the "workaround" for pandas data.  In other words, it seems to me like the pandas-based version should have a "special" name to make it clear that dask-sql is using a work-around for missing functionality.

Reply:
Just to clarify, would that look like moving the original content of `apply_sort`:

https://github.com/dask-contrib/dask-sql/blob/4d5f7dd4c133f7f2c584db2492c8f0ef3cba7e31/dask_sql/physical/utils/sort.py#L15-L38

To its own function that specifies it is a workaround, and then have `apply_sort` do something like:

```python
if isinstance(df, dask_cudf.DataFrame) and not any(sort_null_first) and all(sort_ascending):
    try:
        df = df.sort_values(sort_columns)
        return df.persist()
    except ValueError:
        pass

return workaround_sort_function(...) 
```

Reply:
Yes - That is exactly what I have in mind.  It feels to me like it makes more sense to isolate the "workaround" in a special function, rather than the proper sort_values usage.  However, I'd be happy to defer to @nils-braun if he has a different preference.

Reply:
For now, I'm going to opt for a conservative middle ground by placing the content of `multi_col_sort` into `apply_sort`, since it's only 2 lines. If @nils-braun is okay with the restructuring as you've described, then I'll go ahead and split off dask-sql's sorting implementation into a separate function.